snmp/datadog_checks/snmp/snmp.py

```diff
@@ -273,17 +279,15 @@ def fetch_sysobject_oid(self, config):
         self.log.debug('Returned vars: %s', var_binds)
         return var_binds[0][1].prettyPrint()
 
-    def _profile_for_sysobject_oid(self, sys_object_oid):
-        """Return, if any, a matching profile for sys_object_oid.
-
-        If several profiles match, it will return the longer match, ie the
-        closest one to the sys_object_oid.
-        """
+    def _profiles_for_sysobject_oid(self, sys_object_oid):
+        # type: (str) -> List[str]
+        """Return all profiles that match the given sys_object_id."""
         oids = [oid for oid in self.profiles_by_oid if fnmatch.fnmatch(sys_object_oid, oid)]
-        oids.sort()
+
         if not oids:
             raise ConfigurationError('No profile matching sysObjectID {}'.format(sys_object_oid))
-        return self.profiles_by_oid[oids[-1]]
+
+        return [self.profiles_by_oid[oid] for oid in oids]
```

Comment:
You can refactor that by iterating only once on `profiles_by_oid.items()` above.

Reply:
tru!